File: typescript/base-hmi-types/src/screens/widgets/HmiTextWidgetBase.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiColor, HmiProperty } from "../base.js";
import { HmiWidgetBase } from "./HmiWidgetBase.js";

export abstract class HmiTextWidgetBase extends HmiWidgetBase {
  text?: HmiProperty<HmiMultilingualText>;
  alternateText?: HmiProperty<HmiMultilingualText>;
  readOnly?: HmiProperty<boolean>;
  outputFormat?: HmiProperty<string>;
  formatPattern?: HmiProperty<string>;
  fieldLength?: HmiProperty<number>;
  fitToLargest?: HmiProperty<boolean>;
  resizable?: HmiProperty<boolean>;
  extraHeightOffset?: HmiProperty<number>;
  aboveUpperLimitColor?: HmiProperty<HmiColor>;
  belowLowerLimitColor?: HmiProperty<HmiColor>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiTextWidgetBase.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public abstract class HmiTextWidgetBase : HmiWidgetBase
{
    public HmiProperty<HmiMultilingualText>? Text { get; set; }

    public HmiProperty<HmiMultilingualText>? AlternateText { get; set; }

    public HmiProperty<bool>? ReadOnly { get; set; }

    public HmiProperty<string>? OutputFormat { get; set; }

    public HmiProperty<string>? FormatPattern { get; set; }

    public HmiProperty<int>? FieldLength { get; set; }

    public HmiProperty<bool>? FitToLargest { get; set; }

    public HmiProperty<bool>? Resizable { get; set; }

    public HmiProperty<double>? ExtraHeightOffset { get; set; }

    public HmiProperty<HmiColor>? AboveUpperLimitColor { get; set; }

    public HmiProperty<HmiColor>? BelowLowerLimitColor { get; set; }
}
